csharp/src/SeedLang/Runtime/HeapObject.cs

```diff
@@ -222,12 +270,20 @@ internal partial class HeapObject : IEquatable<HeapObject> {
       return intIndex < 0 ? length + intIndex : intIndex;
     }
 
-    private static string TupleToString(IReadOnlyList<Value> tuple) {
+    private static void CheckKey(Value key) {
+      if (!key.IsNil && !key.IsBoolean && !key.IsNumber && !key.IsString && !key.IsRange &&
+          !key.IsTuple) {
+        throw new DiagnosticException(SystemReporters.SeedRuntime, Severity.Fatal, "", null,
+                                      Message.RuntimeErrorUnhashableType);
```

Comment:
It seems RuntimeErrorUnhashableTypes should also be checked when the dictionary value is initiated. For example, the current SeedPython implementation outputs as below:

```
>>> a = { [1, 2, 3]: 1 }
---------- Source ----------
1     a = { [1, 2, 3]: 1 }
---------- Run ----------
>>> print(a)
---------- Source ----------
1     print(a)
---------- Run ----------
{[1, 2, 3]: 1}
>>> a
---------- Source ----------
1     a
---------- Run ----------
{[1, 2, 3]: 1}
>>> a[[1, 2, 3]]
---------- Source ----------
1     a[[1, 2, 3]]
---------- Run ----------
1     a[[1, 2, 3]]
20220319211523331 Fatal (SeedLang.VM) <> [Ln 1, Col 0 - Ln 1, Col 11] 29: RuntimeErrorUnhashableType
```

While Python3 reports Unhashable key error as early as the value is created:

```
Python 3.9.10 (main, Jan 15 2022, 11:48:04) 
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> a = { [1, 2, 3]: 3 }
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
```

Reply:
Fix it by adding check in the constructor.